pkg/cloudprovider/vsphere/nodemanager_test.go

```diff
@@ -154,3 +155,27 @@ func TestExport(t *testing.T) {
 
 	nm.UnregisterNode(node)
 }
+
+func TestReturnIPsFromSpecificFamily(t *testing.T) {
+	ipFamilies := []string{
+		"10.161.34.192",
+		"fd01:0:101:2609:bdd2:ee20:7bd7:5836",
+		"fe80::98b5:4834:27a8:c58d",
```

Comment:
Suggestion - Have an IPv6 address ending in :: as a test - Have seen this cause problems in parsing in other IPv6 implementations in code.

Example:

Start of addresses in a /64
fd01:0:101:2609::
User is doing random tracking of machine types and starting each machine type ee20 at 0:0
fd01:0:101:2609:bdd2:ee20::
And Simlar to above but last segment.
fd01:0:101:2609:bdd2:ee20:7bd7::

Reply:
I ended up using real addresses based on vsphere config that I have running. I did try this with the addresses you have suggested and it looks good. Thanks for the input!